### ISSUE TITLE
Feature request - implement Adaptive Gradient Clipping and/or Gradient Centralization

### ISSUE THREAD

**Issue by jdart1** [jw1912/bullet#495]
[Adaptive Gradient Clipping](https://arxiv.org/pdf/2106.13731) and [Gradient Centralization](https://arxiv.org/pdf/2004.01461) are techniques that can be added to a variety of optimizers, including AdamW and Ranger, to improve convergence and accuracy of the trained model.  They are quite simple to apply. These among other techniques are used in Google's [Ranger21 optimizer](https://arxiv.org/abs/2106.13731),

**Comment by jw1912** [comment]
Will do after I finish #488
